packages/autocomplete-core/src/__tests__/getAutocompleteSetters.test.ts

```diff
@@ -51,19 +51,58 @@ describe('createAutocomplete', () => {
     });
   });
 
-  test('setSuggestions', () => {
-    const onStateChange = jest.fn();
-    const { setSuggestions } = createAutocomplete({
-      getSources: () => [],
-      onStateChange,
+  describe('setSuggestions', () => {
+    test('default', () => {
+      const onStateChange = jest.fn();
+      const { setSuggestions } = createAutocomplete({
+        getSources: () => [],
+        onStateChange,
+      });
+      const suggestions = [createSuggestion([{ item: 'hi' }])];
+
+      setSuggestions(suggestions);
+
+      expect(onStateChange).toHaveBeenCalledTimes(1);
+      expect(onStateChange).toHaveBeenCalledWith({
+        state: expect.objectContaining({
+          suggestions: expect.arrayContaining(
+            suggestions.map((suggestion) => ({
```

Comment:
Can we not put logic in our expectations but rather inline the result? It will avoid our tests getting out of hand and it'll keep them more understandable.